homeassistant/components/nest/event.py

```diff
@@ -0,0 +1,127 @@
+"""Event platform for Google Nest."""
+
+from dataclasses import dataclass
+import logging
+
+from google_nest_sdm.device import Device
+from google_nest_sdm.device_manager import DeviceManager
+from google_nest_sdm.event import EventMessage, EventType
+from google_nest_sdm.traits import TraitType
+
+from homeassistant.components.event import (
+    EventDeviceClass,
+    EventEntity,
+    EventEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_DEVICE_MANAGER, DOMAIN
+from .device_info import NestDeviceInfo
+from .events import (
+    EVENT_CAMERA_MOTION,
+    EVENT_CAMERA_PERSON,
+    EVENT_CAMERA_SOUND,
+    EVENT_DOORBELL_CHIME,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class NestEventEntityDescription(EventEntityDescription):
+    """Entity description for nest event entities."""
+
+    trait_type: TraitType | None = None
+    api_event_type: EventType | None = None
+    event_type: str | None = None
+    has_entity_name = True
+
+
+ENTITY_DESCRIPTIONS = [
+    NestEventEntityDescription(
+        key=EVENT_DOORBELL_CHIME,
+        translation_key="chime",
+        device_class=EventDeviceClass.DOORBELL,
+        trait_type=TraitType.DOORBELL_CHIME,
+        api_event_type=EventType.DOORBELL_CHIME,
+    ),
+    NestEventEntityDescription(
+        key=EVENT_CAMERA_MOTION,
+        translation_key="motion",
+        device_class=EventDeviceClass.MOTION,
+        trait_type=TraitType.CAMERA_MOTION,
+        api_event_type=EventType.CAMERA_MOTION,
+    ),
+    NestEventEntityDescription(
+        key=EVENT_CAMERA_PERSON,
+        translation_key="person",
+        device_class=EventDeviceClass.MOTION,
+        trait_type=TraitType.CAMERA_PERSON,
+        api_event_type=EventType.CAMERA_PERSON,
+    ),
+    NestEventEntityDescription(
+        key=EVENT_CAMERA_SOUND,
+        translation_key="sound",
+        device_class=EventDeviceClass.MOTION,
+        trait_type=TraitType.CAMERA_SOUND,
+        api_event_type=EventType.CAMERA_SOUND,
+    ),
```

Comment:
Should this be a single event entity that can fire `motino`, `persion`, `sound` ?

Reply:
Updated to use a single entity. I was considering this option and was on the fence between a few and appreciate the input.

The part that i'm still thinking about is how to expose media associated with the events. My current thinking is:
- we want doorbell chime to happen as fast as possible without fetching media to reduce lag
- add an additional event type like "media"  (or even more specific like "image" or "clip") to the doorbell chime event with a media source path attribute.  (This could also be a separate event entity, but keeping on the same entity would be nice)


Then for  motion, person, sound i'm not sure if its worth splitting or not by media.  We could make them only sent with media (like they device triggers are today) or also have separate events for media/image/clip -- but if we split like above it could reduce latency, but it might be harder to use.

Reply:
@joostlek curious for any thoughts you have on this

Reply:
So you want to also send a media url with the event? That could be the state attributes of the entity. I can also remember that some integrations have an image entity that show the latest event image/media thing.

Reply:
Yes agree on the attribute for media url.

Though in the doorbell case in particular I want two events. One for the chime without waiting for media fetch and another with aedia attribute. 

Need to decide if that's a separate event entity or another event type or how the user may be able to keep the events apart. I'm assuming it may be needed for easier automations otherwise they may think the doorbell was pushed twice

Reply:
Hmm, I think two events is strange because the real life event only happens once. So also 2 triggers would be meh.

I would expect either an old school event (but that's also quite meh) or a device trigger of some sort

Reply:
Would it make sense to have triggers for the media browser that can tell us if an entry got added?

Reply:
My takeway from the discussion is your recommendation is actually the status quo and i'll want to make a blueprint that can handle the notification association complexity. In a follow up i'll make the actual events here update faster without waiting for media.

Reply:
I think it would be nice to make nicer, but I don't see any way with entities